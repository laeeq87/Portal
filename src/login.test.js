import React from "react";
import { render, screen, cleanup} from '@testing-library/react';
import Login from "./login";
import { createMemoryHistory } from 'history'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'
import { Router } from "react-router";

afterEach(()=> {
    cleanup();
});

const props = {
    match:{
        params: jest.fn(),
    }
  }

it("Should Render Home Component", ()=> {
    const history = createMemoryHistory(['/', '/login']);
    render(
        <Router history={history}>
            <Login {...props} history={history}/>
        </Router>
    );
    const homeComponentId  = screen.getByTestId('tst-login')
    expect(homeComponentId).toBeInTheDocument();
}); 

it("Should Render Login Button in Home Component", ()=> {
    const history = createMemoryHistory(['/', '/login']);
    render(
        <Router history={history}>
            <Login {...props} history={history}/>
        </Router>
    );
    const loginByttonId = screen.getByTestId('tst-login-btn')
    expect(loginByttonId).toBeInTheDocument();
    expect(loginByttonId).toHaveTextContent('login')
}); 

it("Should Render Email Input Field Login Component", ()=> {
    const history = createMemoryHistory(['/', '/login']);
    render(
        <Router history={history}>
            <Login {...props} history={history}/>
        </Router>
    );
    const loginByttonId = screen.getByTestId('tst-login-input-email')
    expect(loginByttonId).toBeInTheDocument();
    expect(loginByttonId).toHaveTextContent('Email address')
}); 


test('matches snapshot', ()=> {
    const history = createMemoryHistory(['/', '/login']);
    const tree = renderer.create(
        <Router history={history}>
            <Login {...props} history={history}/>
    </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})
