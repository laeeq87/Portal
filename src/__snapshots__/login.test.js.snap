// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="tst-login"
>
  <div
    className="row justify-content-center align-items-center mt-5"
    style={
      Object {
        "height": "70vh",
      }
    }
  >
    <div
      className="col-md-3 pb-5  my-border"
    >
      <div
        className="row justify-content-center align-items-cente"
      >
        <h4
          className="text-center my-5"
        >
          Login
        </h4>
        <div
          className="form-floating mb-3"
        >
          <input
            className="form-control"
            id="floatingInput"
            onChange={[Function]}
            placeholder="name@example.com"
            type="email"
          />
          <label
            data-testid="tst-login-input-email"
            style={
              Object {
                "left": 10,
              }
            }
          >
            Email address
          </label>
        </div>
        <div
          className="form-floating"
        >
          <input
            className="form-control"
            id="floatingPassword"
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />
          <label
            style={
              Object {
                "left": 10,
              }
            }
          >
            Password
          </label>
        </div>
        <div
          className="row justify-content-center align-items-center mt-5"
        >
          <div
            className="col-6 text-center"
          >
            <button
              className="btn btn-primary"
              data-testid="tst-login-btn"
              disabled={true}
              onClick={[Function]}
              style={
                Object {
                  "height": 50,
                  "width": 115,
                }
              }
            >
              login
            </button>
          </div>
        </div>
        <div />
      </div>
    </div>
  </div>
</div>
`;
